document/store/src/main/java/io/camunda/document/store/EnvironmentConfigurationLoader.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.document.store;
+
+import io.camunda.document.api.DocumentStoreConfiguration;
+import io.camunda.document.api.DocumentStoreConfiguration.DocumentStoreConfigurationRecord;
+import io.camunda.document.api.DocumentStoreProvider;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * DOCUMENT_HASH_VERIFICATION_ENABLED=true
+ * DOCUMENT_STORE_GCP_CLASS=io.camunda.document.store.gcp.GcpDocumentStoreProvider
+ * DOCUMENT_STORE_GCP_BUCKET=my-bucket
+ * DOCUMENT_STORE_INMEMORY_CLASS=io.camunda.document.store.inmemory.InMemoryDocumentStoreProvider
+ */
+public class EnvironmentConfigurationLoader implements DocumentStoreConfigurationLoader {
+
+  private static final String CONFIGURATION_PREFIX = "DOCUMENT_";
+  private static final String STORE_PREFIX = "DOCUMENT_STORE_";
+  private static final String DEFAULT_DOCUMENT_STORE_ID = "DEFAULT_STORE_ID";
+
+  private static final Pattern DOCUMENT_STORE_PROPERTY_PATTERN =
+      Pattern.compile("^" + STORE_PREFIX + "(?<id>[^_]+)_(?<property>.+)$");
+  private static final String STORE_CLASS = "CLASS";
+
+  private final Map<String, String> environmentVariableOverrides = new HashMap<>();
+
+  @Override
+  public DocumentStoreConfiguration loadConfiguration() {
+    return new DocumentStoreConfiguration(
+        getRootLevelProperty(DEFAULT_DOCUMENT_STORE_ID).map(String::toLowerCase).orElse(null),
+        loadDocumentStoreProperties());
+  }
+
+  // For testing purposes
+  void addEnvironmentVariableOverride(final String key, final String value) {
```

Comment:
🔧 I'd prefer to remove this method from the code. You can inject the map in the constructor instead, or you can use the system properties.

If you choose the second approach you need to modify the `getEnvVariable()` with a fallback on the system properties, like:

```
private Optional<String> getEnvVariable(final String name) {
  return Optional.ofNullable(environmentVariableOverrides.get(name))
      .or(() -> Optional.ofNullable(System.getenv(name)))
      .or(() -> Optional.ofNullable(System.getProperty(name)));
}
```